src/qibocal/auto/history.py

```diff
@@ -1,70 +1,25 @@
 """Track execution history."""
-import copy
-import os
-from dataclasses import dataclass
-from functools import cached_property
-from pathlib import Path
-from typing import Optional
 
-from .operation import Data, Results
 from .runcard import Id
-from .status import Status
-from .task import Task
+from .task import Completed
 
 
-@dataclass
-class Completed:
-    """A completed task."""
+def add_timings_to_meta(meta, history):
```

Comment:
```suggestion
def add_timings(meta, history):
```

Not my favorite name even like this, but I just wanted to remark conciseness once more :)

Being `meta` is chance, and it's already in the named of the variable both in the function and call site - the purpose of the function is just to add the timings to the parameter passed, whatever it is (provided it is a dictionary, of course).

Reply:
Btw, I'd avoid getting meta as input: just create a new dictionary with the information.

If needed, you can merge them after the call.

There is a [dedicated operator](https://docs.python.org/3/library/stdtypes.html?highlight=dict#dict.values) to merge dictionaries in Python:
```py
merged = d1 | d2
```

Up to py3.8 the proper way would have been `{**d1, **d2}`

More info about merging dicts:
https://stackoverflow.com/a/26853961/8653979

Reply:
I see your point. There could be a few problems with just passing a dictionary.
Given that by default `data_time` and `results_time` are 0, I might overwrite a value in meta with 0 while it already had a value (for instance when running just fitting it will overwrite the acquisition time to 0). Which is the reason for each I implemented several checks below. 

A guess that I could have two separate functions. One for creating the timing dict and a second one for combining the current meta with the timing dict.